### ISSUE TITLE
runtime: SetCgoTraceback drops frames from tracebacks

### ISSUE THREAD

**Issue by mpx** [golang/go#29034]
go version devel +2140975ebd Thu Nov 29 22:23:02 2018 +0000 linux/amd64

Using `runtime.SetCgoTraceback` along with `github.com/ianlancetaylor/cgosymbolizer` to record profiles that span Go and C code causes the bottom frames from tracebacks to be lost. Each additional C function on the stack causes another function frame to be lost from the bottom of the traceback.

Example:
```go
package main

import (
    "os"
    "runtime/pprof"

    _ "github.com/ianlancetaylor/cgosymbolizer"
)

/*
#cgo CFLAGS: -O0

void run() {
    int i;
    for (i = 0; i < 1000000000; i++);
}
*/
import "C"

func main() {
    f, _ := os.Create("cpu.prof")
    defer f.Close()
    pprof.StartCPUProfile(f)
    C.run()
    pprof.StopCPUProfile()
    f.Close()
}
```

Running with `go build && ./example && go tool pprof -traces cpu.prof` gives the following:
```
run   
_cgo_e4e719602ade_Cfunc_run
runtime.asmcgocall
runtime.cgocall
main._Cfunc_run
```

`main.main` and `runtime.main` have been lost from the bottom of the trace.

**Comment by mpx** [comment]
Cc @ianlancetaylor

**Comment by andybons** [comment]
@mpx is this a regression from 1.11?

**Comment by mpx** [comment]
No, I originally found the bug in 1.11.

**Comment by mpx** [comment]
Without this fix CGO profiles can be subtly confusing/unhelpful since frames are lost. The bug and the fix are fairly trivial:

```go
var stk [32]uintptr
cgoOff := copy(stk[:], cStk)
n := copy(stk[cgoOff:], goStk)
// Missing fix: n += cgoOff
frames := stk[:n]
```

I understand the release is getting close and the need to reduce change risk, but could this fix please be considered for 1.12? Is this bug/fix trivial enough to make it in?
